Description: CMake: Fix QObject::connect failing on ARM

 We need PIE, doesn't matter if reduce_relocations is used or not
Author: Albert Astals Cid <albert.astals@canonical.com>

Index: qtbase-opensource-src-5.4.0+dfsg/mkspecs/features/create_cmake.prf
===================================================================
--- qtbase-opensource-src-5.4.0+dfsg.orig/mkspecs/features/create_cmake.prf
+++ qtbase-opensource-src-5.4.0+dfsg/mkspecs/features/create_cmake.prf
@@ -171,8 +171,6 @@ contains(CONFIG, plugin) {
     return()
 }
 
-unix:contains(QT_CONFIG, reduce_relocations):CMAKE_ADD_FPIE_FLAGS = "true"
-
 CMAKE_MKSPEC = $$[QMAKE_XSPEC]
 
 CMAKE_MODULE_DEPS = $$cmakeModuleList($$sort_depends(QT.$${MODULE}.depends, QT.))
Index: qtbase-opensource-src-5.4.0+dfsg/src/corelib/Qt5CoreConfigExtras.cmake.in
===================================================================
--- qtbase-opensource-src-5.4.0+dfsg.orig/src/corelib/Qt5CoreConfigExtras.cmake.in
+++ qtbase-opensource-src-5.4.0+dfsg/src/corelib/Qt5CoreConfigExtras.cmake.in
@@ -66,15 +66,13 @@ list(APPEND Qt5Core_INCLUDE_DIRS ${_qt5_
 set_property(TARGET Qt5::Core APPEND PROPERTY INTERFACE_INCLUDE_DIRECTORIES ${_qt5_corelib_extra_includes})
 set(_qt5_corelib_extra_includes)
 
-!!IF !isEmpty(CMAKE_ADD_FPIE_FLAGS)
 # Targets using Qt need to use the POSITION_INDEPENDENT_CODE property. The
 # Qt5_POSITION_INDEPENDENT_CODE variable is used in the # qt5_use_module
 # macro to add it.
 set(Qt5_POSITION_INDEPENDENT_CODE True)
 set(Qt5Core_EXECUTABLE_COMPILE_FLAGS \"-fPIC\")
 set_property(TARGET Qt5::Core PROPERTY INTERFACE_POSITION_INDEPENDENT_CODE \"ON\")
 set_property(TARGET Qt5::Core APPEND PROPERTY INTERFACE_COMPILE_OPTIONS ${Qt5Core_EXECUTABLE_COMPILE_FLAGS})
-!!ENDIF
 
 !!IF !isEmpty(QT_NAMESPACE)
 list(APPEND Qt5Core_DEFINITIONS -DQT_NAMESPACE=$$QT_NAMESPACE)
